### DISCUSSION TITLE
State Recovery

### DISCUSSION THREAD

**Discussion by spencerTaira** [miguelgrinberg/Flask-SocketIO#1987] [Q&A] [answered]
Hi,

I ran into an issue when trying to handle websocket reconnections. When reading the SocketIO docs, I read about state recovery upon reconnection, but I can't get it to work with Flask SocketIO.

Is this supported on Flack SocketIO? And if so, can I please get some help with implementing it? I am using a React front-end with socketio.

**Comment by miguelgrinberg** [comment, marked as answer]
No, that option is not implemented in this server.

**Reply by spencerTaira** [reply]
Thanks

**Comment by Pratster95** [comment]
@miguelgrinberg If my server is built using Flask Socket IO and Client using Javascript then how do I implement Connection State Recovery on the server side?

**Reply by miguelgrinberg** [reply]
It really depends on the application. In a lot of applications the same logic that creates the state on the first connection can be used in reconnections. Other applications may choose to store state in Redis, so that it can survive a client disconnection. It really depends, there's a lot of ways to do this.

**Reply by Pratster95** [reply]
@miguelgrinberg Thanks for answering this. I am using redis to store the session id.  Are we also sending any offset from the server side in this case? https://socket.io/docs/v4/tutorial/step-7

**Reply by miguelgrinberg** [reply]
I'm not sure I understand your question. You are looking at a tutorial for the JavaScript server. This is the Python server. It is not the same. As I said above, the Python server does not implement state recovery at this time.

**Reply by Pratster95** [reply]
Yes. Since there is no implementation on Python Server today, when we store the session id in redis at the server side, do we use the same implementation of connectionStateRecovery on the client side (Javascript)

**Reply by miguelgrinberg** [reply]
I'm sorry, but I do not understand. This could be because I'm not very familiar with the connection state recovery feature. My understanding is that this is a server feature that restores the rooms and the user session upon reconnection. I'm not sure what you mean by "connectionStateRecovery on the client side". I don't think this is a feature that is used in the client.
